python/neuroglancer/ingest/volumes.py

```diff
@@ -55,85 +65,66 @@ def underlying(self):
 
     @property
     def num_channels(self):
-        if len(self.shape) == 3:
+        if len(self._shape) == 3:
             return 1
-        elif len(self.shape) == 4:
-            return self.shape[0]
+        elif len(self._shape) == 4:
+            return self._shape[-1]
         else:
-            raise Exception('Wrong shape')  
+            raise Exception('Wrong shape')
+
+    @property
+    def encoding(self):
+        return 'raw'
+      
     
 class HDF5Volume(Volume):
 
-    def __init__(self, path):
-        import h5py
+    def __init__(self, path, layer_type):
         self._f = h5py.File(path, 'r')
         self._data = self._f['main']
-        self._layer_type = 'image'
-        self._mesh = False
-        self._resolution = [4,4,40]
-        self._underlying = self.shape
-        self._data_type = self._f['main'].dtype
 
-    @property
-    def shape(self):
-        return self._data.shape[::-1]
+        self._layer_type = layer_type
+        
+        if self._layer_type == 'image':
+            self._mesh = False
+        elif self._layer_type == 'segmentation':
+            self._mesh = True
+        elif self._layer_type == 'affinities':
+            self._mesh = False
 
+        self._resolution = [6,6,30]
+        self._shape = self._data.shape[::-1]
+        self._underlying = self.shape
+        self._data_type = self._f['main'].dtype
 
     def __getitem__(self, slices):
         """
         Asumes x,y,z coordinates
         """
-        return np.swapaxes(self._data.__getitem__(slices[::-1]),0,2)
+        data = self._data.__getitem__(slices[::-1])
+        if self._layer_type == "affinities":
+            return data.transpose((3,2,1,0))
+        else:
+            return np.expand_dims(np.swapaxes(data,0,2),3)
 
     def __del__(self):
         self._f.close()
 
-class FakeVolume(Volume):
+
+class NumpyVolume(Volume):
 
     def __init__(self):
         arr = np.ones(shape=(127,127,127),dtype=np.uint32)
         self._data = np.pad(arr, 1, 'constant')
-        self._layer_type = 'image'
-        self._mesh = False
+        self._layer_type = 'segmentation'
+        self._mesh = True
         self._resolution = [6,6,30]
         self._underlying = self.shape
         self._data_type = self._data.dtype
-
-    @property
-    def shape(self):
-        return self._data.shape
-
+        self._shape = self._data.shape
 
     def __getitem__(self, slices):
         """
         Asumes x,y,z coordinates
         """
         return self._data.__getitem__(slices)
-
-class DVIDVolume(Volume):
```

Comment:
It doesn't look like we will use this in the near future. If I get proved wrong, we can always come back to the history an restore it.